modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc

```diff
@@ -0,0 +1,361 @@
+      'attributes' => array('class' => array('add-denomination', 'button'))
+    ));
+
+    // Now, get the denominations for this denomination type.
+    $denominations = commerce_pos_currency_denomination_get_denominations(array($denomination_type_info->currency_code));
+    if (!empty($denominations)) {
+      $output .= t('<h3>Current Denominations for Type @type</h3>', array(
+        '@type' => $denomination_type_info->currency_code
+      ));
+
+      $output .= '<table class="commerce-pos-current-denominations">';
+      $output .= '<tr>';
+      $output .= '<th>' . t('Title') . '</th>';
+      $output .= '<th>' . t('Amount') . '</th>';
+      $output .= '<th>' . t('Options') . '</th>';
+      $output .= '</tr>';
+
+      foreach ($denominations[$denomination_type_info->currency_id] as $denomination) {
+        $output .= '<tr>';
+        $output .= '<td>' . $denomination->denomination_title . '</td>';
+        $output .= '<td>' . $denomination->denomination_amount . '</td>';
+        $output .= '<td>' . l(t('Delete Denomination'), 'admin/commerce/config/pos/denomination-type/' . $currency_id . '/denomination/' . $denomination->denomination_id . '/delete') . '</td>';
+        $output .= '</tr>';
+      }
+
+      $output .= '</table>';
+    }
+    else {
+      $output .= t('<p>Currently, no denominations have been added for this denomination type.</p>');
+    }
+  }
+  else {
+    $output .= t('<p>Invalid denomination type.</p>');
+  }
+
+  return $output;
+}
+
+/**
+ * The form for adding a new denomination.
+ */
+function commerce_pos_currency_add_denomination_form($form, &$form_state, $currency_id) {
+  $form_state['currency_id'] = $currency_id;
+
+  // Get the denomination info for this type.
+  $form_state['denomination_type_info'] = commerce_pos_denomination_get_currency_types_from_id(array($currency_id));
+
+  if (!empty($form_state['denomination_type_info'])) {
+    $form_state['denomination_type_info'] = $form_state['denomination_type_info'][$currency_id];
+
+    drupal_set_title(t('Add Denomination for Type @type', array(
+      '@type' => $form_state['denomination_type_info']->currency_code
+    )));
+
+    $form['commerce_pos_denomination_name'] = array(
+      '#type' => 'textfield',
+      '#size' => 25,
+      '#maxlength' => 25,
+      '#default_value' => NULL,
+      '#title' => t('Denomination Name'),
+      '#description' => t('Enter the denomination name eg. "penny".'),
+      '#required' => TRUE,
+    );
+
+    $form['commerce_pos_denomination_title'] = array(
+      '#type' => 'textfield',
+      '#size' => 50,
+      '#maxlength' => 50,
+      '#default_value' => NULL,
+      '#title' => t('Denomination Title'),
+      '#description' => t('Enter the denomination title eg. "Penny".'),
+      '#required' => TRUE,
+    );
+
+    $form['commerce_pos_denomination_amount'] = array(

```

Comment:
Consider using an element_validate property here to validate that they enter a positive integer.